internal/v4/relations.go

```diff
@@ -222,8 +227,12 @@ func (h *Handler) metaBundlesContaining(entity *mongodoc.Entity, id *charm.Refer
 		if err != nil {
 			return nil, errgo.Notef(err, "cannot retrieve bundle metadata")
 		}
+		url := e.URL
+		if e.PromulgatedURL != nil {
```

Comment:
I've seen this quite a bit.
How about a method on Entity?

  // PreferredURL returns the preferred form of the URL to
  // to refer to the given entity.
  func (e *Entity) PreferredURL() *charm.Reference {
         if e.PromulgatedURL != nil { 
             return e.PromulgatedURL
         }
         return e.URL
     }